test/Microsoft.AspNetCore.Server.KestrelTests/ConnectionTests.cs

```diff
@@ -0,0 +1,36 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Server.Kestrel;
+using Microsoft.AspNetCore.Server.Kestrel.Http;
+using Microsoft.AspNetCore.Server.Kestrel.Infrastructure;
+using Microsoft.AspNetCore.Server.Kestrel.Networking;
+using Microsoft.AspNetCore.Server.KestrelTests.TestHelpers;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Server.KestrelTests
+{
+    public class ConnectionTests
+    {
+        [Fact]
+        public void DoesNotEndConnectionOnZeroRead()
+        {
+            using (var memory = new MemoryPool2())
+            {
+                var trace = new TestKestrelTrace();
+                var mockLibuv = new MockLibuv();
+                var context = new ListenerContext(new TestServiceContext())
+                {
+                    FrameFactory = connectionContext => new Frame<HttpContext>(
+                        new DummyApplication(httpContext => TaskUtilities.CompletedTask), connectionContext),
+                    ServerAddress = ServerAddress.FromUrl("http://localhost:5000")
+                };
+                var socket = new MockSocket(mockLibuv, trace);
+                var connection = new Connection(context, socket);
+                connection.Start();
+
+                Libuv.uv_buf_t ignored = new Libuv.uv_buf_t();
+                mockLibuv.ReadCallback(socket.InternalGetHandle(), 0, ref ignored);
```

Comment:
Can we contain the `InternalGetHandle` ugliness to MockLibuv like we do with OnWrite?